controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -3026,46 +3026,53 @@ bool master_thread::handle_cmdu_control_message(const std::string &src_mac,
             database.update_node_last_seen(client_mac);
         }
 
-        if (database.get_node_type(client_mac) == beerocks::TYPE_CLIENT) {
-            auto db_ipv4 = database.get_node_ipv4(client_mac);
-            if (!database.set_node_ipv4(client_mac, ipv4)) {
-                LOG(ERROR) << "set node ipv4 failed";
-            }
+        if (database.get_node_type(client_mac) != beerocks::TYPE_CLIENT) {
+            LOG(DEBUG) << "node " << client_mac << " is not of TYPE_CLIENT";
+            return true;
+        }
 
-            if (!database.set_node_name(
-                    client_mac, std::string(notification_in->name(message::NODE_NAME_LENGTH)))) {
-                LOG(ERROR) << "set node name failed";
-            }
+        if (!database.set_node_ipv4(client_mac, ipv4)) {
+            LOG(ERROR) << "set node ipv4 failed";
```

Comment:
Does it makes sense to continue in this case?
If you don't have an IP, what are you actually sending to the agent?

Reply:
Actually it cannot be failed if the previous db setter success.
The set ipv4 failed if the client has not been found on the db.
I'll remove the failure handling.